backend-tests/tests/test_inventory.py

```diff
@@ -311,3 +315,579 @@ def test_fail_no_attr_value(self, user):
                 "PATCH", inventory.URL_DEVICE_ATTRIBUTES, payload
             )
             assert r.status_code == 400
+
+
+class TestDeviceFilteringEnterprise:
+    @property
+    def logger(self):
+        try:
+            return self._logger
+        except AttributeError:
+            self._logger = logging.getLogger(self.__class__.__name__)
+        return self._logger
+
+    def is_subset(self, value, subset):
+        if type(value) != type(subset):
+            self.logger.error("type(%s) != type(%s)" % (value, subset))
+            return False
+
+        if isinstance(subset, list):
+            for i, item in enumerate(subset):
+                if not self.is_subset(value[i], item):
+                    return False
+
+        elif isinstance(subset, dict):
+            for k, v in subset.items():
+                if k not in value:
+                    self.logger.error("%s not in %s" % (k, list(value.keys())))
+                    return False
+                elif not self.is_subset(value[k], v):
+                    return False
+
+        elif value != subset:
+            self.logger.error("%s != %s" % (value, subset))
+            return False
+
+        return True
+
+    def dict_to_inventoryattrs(self, d, scope=None):
+        attr_list = []
+        for key, value in d.items():
+            attr = {"name": key, "value": value}
+            if scope is not None:
+                attr["scope"] = scope
+            attr_list.append(attr)
+
+        return attr_list
+
+    def test_search_v2(self, tenants_users):
```

Comment:
I meant to say it would be useful to test the 'plan-too-low' scenario ("os").
but actually tenants_users does create tenants with this plan.

so something is wrong - they should not be able to use the filtering api, it's PRO and ENT only. I believe KJ submitted some late patches yesterday, fixing just that (used to allow "os").

Reply:
https://github.com/mendersoftware/inventory-enterprise/pull/13

Reply:
Good point. I tested it before this was merged, I'll make a test-case for it.